src_Core/RegFiles/CSR_RegFile_MSU.bsv

```diff
@@ -40,6 +40,15 @@ import ClientServer :: *;
 
 import GetPut_Aux :: *;
 
+`ifdef PERFORMANCE_MONITORING
+import PerformanceMonitor :: *;
+
+`define ctrs 29
+`define evts 63
+`else
+`define ctrs 0
```

Comment:
Good points, the argument for define is that it can be set in the Makefile (with the following change:)
```
`ifndef ctrs
`define ctrs 29
`endif
```
This doesn't make sense for the number of events though which should be fixed and I'll put that in ISA_Decls.
Is it better to choose the number of counters from the Makefile or by changing the code in ISA_Decls?

Reply:
Hm, that's a good question. For now I'd leave it as just a hard-coded typedef in ISA_Decls, and if we decide we need it to be more configurable than on/off we can always add a macro for it (ie change the typedef from `typedef Foo 29;` to ``typedef Foo `FOO`` or similar and not have to touch anything else).

Reply:
I think this is the only real issue that will be messy no matter what. The problem is caused by having a synthesize boundary on the Performance counting module itself ([being added to BlueStuff](https://github.com/CTSRD-CHERI/BlueStuff/pull/2/files)) which means that all of these values must be constants. Therefore even though we know the number of events at the type level (from the various Vectors which we append) we must still give it to the code as an explicit number somewhere. In Flute this can be in ISA_Decls, but for the BlueStuff module this has to either be defined in the Makefile or by requiring Flute (or any other core using the module) to have an explicit config file with these values defined (to then `include). And since this number must match up between Flute and the BlueStuff module it might be easier to take it in the same way in both (be that defined in Makefile or `include from file).
I can't see a simpler way to do this...